cc_proxy.go

```diff
@@ -127,17 +130,44 @@ func (p *ccProxy) connectProxy(proxyURL string) (*client.Client, error) {
 	return client.NewClient(conn), nil
 }
 
+// start is the proxy start implementation for ccProxy.
+func (p *ccProxy) start(pod Pod) (int, string, error) {
+	if pod.config == nil {
+		return -1, "", fmt.Errorf("Pod config cannot be nil")
+	}
+
+	config, ok := newProxyConfig(*(pod.config)).(CCProxyConfig)
+	if !ok {
+		return -1, "", fmt.Errorf("Wrong proxy config type, should be CCProxyConfig type")
+	}
+
+	if config.Path == "" {
+		return -1, "", fmt.Errorf("Proxy path cannot be empty")
+	}
+
+	// construct the socket path the proxy instance will use
+	socketPath := filepath.Join(runStoragePath, pod.id, "proxy.sock")
+	uri := fmt.Sprintf("unix://%s", socketPath)
+
+	args := []string{config.Path, "-uri", uri}
```

Comment:
Maybe we could modify the process comm here by using something like `"proxy-%s", pod.ID`.

Reply:
We already have `pod.id` in the path as the socket is placed into the pod-specific directory. We could also add the pod ID to the socket name, but if we do, we'd be getting very close to the 107 byte socket name limit.